configs/www/init.sls

```diff
@@ -212,9 +212,17 @@ get composer:
 
 # create a named vendor directory that won't clobber composer use in a project-specific repo
 # let the project overlay use the default "vendor" dir
+/var/www/html/api/vendor-lebp-stack-create:
```

Comment:
So this is not the directory that it's created but it's the name of the id, the directory created is the `/var/www/html/api/vendor-lebp-stack` the thing is that I cannot run two file commands on the same id. So I had to run `file.directory` first to make sure that the folder is created and then run the `file.recurse` with the source so it is available. In my testing this worked properly even if I deleted the vendor-lebp-stack folder it got created and composer installed properly

Reply:
From the syntax here, and the fact that you haven't supplied another directory, I'm 99.9% sure this is going to default to the primary behavior and create the directory specified in the name, i.e. vendor-lebp-stack-create. 

Are you having problems with `/var/www/html/api/vendor-lebp-stack` getting created in your environment? I'm not. If you are, that's a bug to fix.

My issues are with `salt://www/files/vendor-lebp-stack` (i.e. the source files) not being found. Because there aren't any source files in `vendor-lebp-stack` if the composer script isn't being run. I'm surprised you're not seeing this error. I bet this is happening because the composer script populates those source directories, so if you haven't actually gone back to a fresh clone of the repo, you probably have the source files, so you haven't seen the issue. The test would be, go back to whatever your "fresh" state is (probably vagrant destroy?) and do a `ls` in `lebp-stack/configs/www/files`. If the directory `vendor-lebp-stack` is there, then you are not in the first-time state. You might be able to do a git reset with `hard` instead of doing a full rm -rf and a fresh clone. 🤷